### PR TITLE
Avoid using `infiniteloop` in `train_cifar10_ddp.py`

### PR DESCRIPTION
**Pull request by Xiaoming-Zhao**
## What does this PR do?

This PR avoids using infinite generator provided by `infiniteloop` and directly use the `dataloader` instead as discussed in #144. This change follows [the structure provided by `pytorch`](https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler).

I tested the change locally and make sure that it can run smoothly.

I also added a `--standalone` command line argument in README, without which I cannot make the script run. This argument is also provided by [the official example for single node usage](https://pytorch.org/docs/stable/elastic/run.html#usage).

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **test your PR locally** with `pytest` command?
- [ ] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/images/cifar10/train_cifar10_ddp.py [hunk 1]
```diff
@@ -156,46 +154,48 @@ def train(rank, total_num_gpus, argv):
...
+                        {
+                            "net_model": net_model.state_dict(),
+                            "ema_model": ema_model.state_dict(),
+                            "sched": sched.state_dict(),
+                            "optim": optim.state_dict(),
+                            "step": global_step,
+                        },
+                        savedir
+                        + f"{FLAGS.model}_cifar10_weights_step_{global_step}.pt",
+                    )
```
Comment (Xiaoming-Zhao) [inline examples/images/cifar10/train_cifar10_ddp.py:198]: This chunk seems to change a lot but it actually only modifies where `x1` is read. The chunk comes from the indentation change as we do not need `step_pbar` now.

**Comment by ImahnShekhzadeh** [comment]
Hi @Xiaoming-Zhao,

Thanks for this PR!

1.) I'm pretty sure that using `infiniteloop` is correct, and hence does not *need* to be replaced (though that doesn't mean we cannot do it). In issue #144, you wrote:

> I also noticed that the use sampler.set_epoch(epoch). Based on my previous experience, this is crucial to ensure randomness across epochs. However, with he current generator provided by infiniteloop, I am not sure whether the set_epoch will actually affect the dataloader , I need to double check.

I wrote a small script that I uploaded on my website (https://imahnshekhzadeh.github.io/#Bloghttps://imahnshekhzadeh.github.io/#Blog), and with which I experimented extensively! On a node with at least 2 GPUs, I ran it like this:
```bash
torchrun --nproc_per_node=NUM_GPUS_YOU_HAVE test_inf_loop.py --master_addr [...] --master_port [...]
# torchrun --nproc_per_node=2 test_inf_loop.py --master_addr [...] --master_port [...]
```
When `sampler.set_epoch(epoch)` is used, then we observe:
```bash
Rank: 1, Epoch: 0, Batch: 2, Data:
[tensor([[-1.3042, -1.1097],
        [-0.1320, -0.2751]])]

Rank: 1, Epoch: 1, Batch: 2, Data:
[tensor([[-0.1752,  0.6990],
        [-0.2350,  0.0937]])]
```
So clearly, `sampler.set_epoch(epoch)` does its job! However, when commenting out the two lines, we see this in the output:
```
Rank: 1, Epoch: 0, Batch: 2, Data:
[tensor([[-1.3042, -1.1097],
        [-0.1320, -0.2751]])]

Rank: 1, Epoch: 1, Batch: 2, Data:
[tensor([[-1.3042, -1.1097],
        [-0.1320, -0.2751]])]
```
Clearly, no shuffling happened!

2.) About the `--standalone` flag: Can you please open a separate PR for this? Thanks!

**Comment by Xiaoming-Zhao** [comment]
Thanks for the detailed example, @ImahnShekhzadeh! This is incredibly helpful. Lessons learned.

I will close this PR for now as it seems like all required changes have been implemented in #116.

Regarding the `torchrun` command, I create a separate PR in #149
